### DISCUSSION TITLE
wlogout

### DISCUSSION THREAD

**Discussion by bobblkabb** [openSUSE/openSUSEway#66] [Ideas]
Is there anything similar to wlogout?
Running Leap 15.4.

**Comment by jubalh** [comment]
Someone has it in their home project: https://build.opensuse.org/package/show/home:AndnoVember:CuboCore/wlogout
Maybe ask them to submit to X11:Wayland if they think it's working well.

**Comment by jubalh** [comment]
But looking at https://github.com/ArtsyMacaw/wlogout I see there were no commits since 2 years. And some pull requests are open but unanswered. Same about issues.

**Comment by denisok** [comment]
yeah, looks like project is abandoned. Don't know about UI based menu like that, but what we could do probably is to have it as a menu with `wofi`:
https://hg.sr.ht/~scoopta/wofi/browse/man/wofi.7?rev=tip
with `dmenu` we could have prepared list of actions and some script that would execute them.

It probably would looks similar to:
https://github.com/okraits/rofi-tools

and then binding that to some hot hey, or/and adding it to waybar as icon

**Comment by Albireo67** [comment]
Hey guys,

I know it's an old discussion, but I'm just catching up with the project !

Based on @denisok  suggested approach, I've amended a rofi script that I had around and make a small script for wofi (I'd propose to be saved as ~/.config/wofi/scripts/exit-menu.sh) to launch a small menu with exit options, using the same theme/style as the default wofi launcher. Gist code snippet here:  https://gist.github.com/Albireo67/71d867b65d1be1787b53e196caa6c184

To make it a bit smaller in size, I also added an exit-config file which is called from the script above (see  proposed location there):
https://gist.github.com/Albireo67/c035c7d0eca377fbf5c352643013ae33

Nothing fancy though, only Unicode glyphs in the options labels. Easy to add on a keybind as well, but remember to make it executable.

Everything seems to be working  fine in my testing implementation, though I´d appreciate any comments/feedback about it.

Best regards

**Comment by Albireo67** [comment]
Actually, I can simply embed the config options into the script as variables, so it is 'self-contained'.

In this version: https://gist.github.com/Albireo67/0d47cfe2bcd80f45fda1183404a56171

- No need to refer to `exit-config` file any more. Wofi options are now variables `conf1` and `conf2`
- Confirm dialog is smaller (height=10%), and sorted alphabetically ('No' is listed first), so hitting 'Yes' is now more intentional.

Please, share your comments if you try this out as an exit menu.

**Comment by FilippoBonazziSUSE** [comment]
This was done another way in #90, and I think we are happy with that solution as a default for now. More can always be left to user configuration.
